frontend/src/components/SettingsCalendar.vue

```diff
@@ -274,75 +286,101 @@ onMounted(async () => {
       <secondary-button
         :label="t('label.addCalendar', { provider: t('label.google') })"
         class="btn-add text-sm !text-teal-500"
-        @click="addCalendar(CalendarProviders.Google)"
+        @click="discoverGoogleCalendars"
         :disabled="inputMode"
         :title="t('label.addCalendar', { provider: t('label.google') })"
       />
       <secondary-button
         :label="t('label.addCalendar', { provider: t('label.caldav') })"
         class="btn-add text-sm !text-teal-500"
-        @click="addCalendar(CalendarProviders.Caldav)"
+        @click="discoverCaldavCalendars"
         :disabled="inputMode"
         :title="t('label.addCalendar', { provider: t('label.caldav') })"
       />
     </div>
 
     <!-- CalDAV calendar discovery -->
-    <div class="hidden flex-col gap-6">
-      <div class="text-lg">Discover CalDAV Calendars</div>
-      <div class="flex max-w-2xl flex-col gap-4 pl-6">
-        <label class="mt-4 flex items-center pl-4">
-          <div class="w-full max-w-2xs">principal</div>
-          <input
-            v-model="principal.url"
-            type="text"
-            class="w-full max-w-sm rounded-md"
-          />
-        </label>
-        <label class="flex items-center pl-4">
-          <div class="w-full max-w-2xs">{{ t('label.username') }}</div>
-          <input
-            v-model="principal.user"
-            type="text"
-            class="w-full max-w-sm rounded-md"
+    <div v-if="discoveryMode" class="flex max-w-2xl flex-col gap-4 pl-6">
+      <div class="text-lg">{{ t('heading.discoverCaldavcalendars') }}</div>
+      <alert-box
+        v-if="caldavDiscoverySuccess"
+        :scheme="AlertSchemes.Success"
+        title="CalDAV discovery success"
+        @close="caldavDiscoverySuccess = ''"
+      >{{ caldavDiscoverySuccess }}</alert-box>
+      <alert-box
+        v-if="caldavDiscoveryError"
+        title="CalDAV discovery Error"
+        @close="caldavDiscoveryError = ''"
+      >{{ caldavDiscoveryError }}</alert-box>
+      <div class="flex flex-col gap-4">
+        <form
+          class="flex max-w-2xl flex-col gap-4 pl-6"
+          ref="caldavFormRef"
+          @submit.prevent
+          @keyup.enter="() => getRemoteCaldavCalendars()"
+        >
+          <label class="mt-4 flex items-center gap-2 pl-4">
+            <div class="w-full max-w-2xs">
+              {{ t('label.principal') }}
+              <div class="text-xs text-gray-500">{{ t('calDAVForm.help.location') }}</div>
+            </div>
+            <input
+              v-model="principal.url"
+              type="url"
+              class="w-full max-w-sm rounded-md"
+              :disabled="processPrincipal"
+              required
```

Comment:
If you don't mind adding some `data-testid` attributes to your new elements please that'd be great, in case future automated E2E tests use these new fields. If you search the source for other `data-testid` attributes you'll see the naming convention, I've been trying to follow something like `<ui-location-description>-<element-description>-<element-type>`.

i.e. Maybe in this case could be something like:

`data-testid="settings-calendar-caldav-url-input"`

And for the other new buttons and input fields too would be great. Thanks! :)

Reply:
Of course, will do! (sorry, always forgetting about that 😅 )

Reply:
Added!